button-card.js

```diff
@@ -121,118 +127,173 @@
       return retval;
     }
 
-    buildCssColorAttribute(state, config) {
-      let color = config.color;
-      if (state) {
-        let configState = this.testConfigState(state, config);
-        if (configState) {
-          color = configState.color ? configState.color : config.color_off;
-          if (configState.color === 'auto') {
-            color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : configState.default_color;
-          }
+    buildCssColorAttribute(state, config, configState) {
+      let colorValue = null;
+      if (configState && configState.color) {
+        colorValue = configState.color;
+      } else {
+        if (config.color != 'auto' && state && state.state == 'off')
+          colorValue = config.color_off;
+        else
+          colorValue = config.color;
+      }
+      if (colorValue == 'auto') {
+        if (state) {
+          const color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : config.default_color;
+          return color;
         } else {
-          if (config.color === 'auto') {
-            color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : config.default_color;
+          return config.default_color;
+        }
+      } else {
+        if (!colorValue) {
+          if (state) {
+            if (state.state == 'off') {
+              const color = config.color_off;
+              return color;
+            } else {
+              const color = config.default_color;
+              return color;
+            }
+          } else {
+            const color = config.default_color;
+            return color;
           }
-          color = state.state === 'on' ? color : config.color_off;
+        } else {
+          const color = colorValue;
+          return color;
```

Comment:
why set `color` to just return it right away? Either return it once at the end or remove the const all together

Reply:
I would personally prefer a `color` variable defined at the top and returned at the bottom